game-core/src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -41,6 +41,7 @@
  *
  * @param <T> The type of the setting value.
  */
+@SuppressWarnings("StaticInitializerReferencesSubClass")
```

Comment:
I saw this warning too, is it safe to ignore?

Reply:
The class is plastered with these warnings : (
We've had this configuration and warnings for a long time. If it's a problem, it'll be a problem whether we ignore the warnings or not. There is a practicality for ignoring the warnings so we don't mask other warnings.

The text from the analysis describes this problem as a potential threading problem:

> Reports classes that refer to their own subclasses in their static initializers or in static fields. Such references can cause JVM-level deadlocks in multithreaded environment, when one thread tries to load superclass and another thread tries to load subclass at the same time.